pkg/aws/region/flag.go

```diff
@@ -24,18 +24,29 @@ import (
 	"github.com/spf13/pflag"
 
 	"github.com/openshift/rosa/pkg/helper"
+	"github.com/openshift/rosa/pkg/reporter"
 )
 
-// AddFlag adds the debug flag to the given set of command line flags.
+var regionFlag = "region"
+
+// AddFlag adds the region flag to the given set of command line flags.
 func AddFlag(flags *pflag.FlagSet) {
 	flags.StringVar(
 		&region,
-		"region",
+		regionFlag,
 		"",
 		"Use a specific AWS region, overriding the AWS_REGION environment variable.",
 	)
 }
 
+// DeprecateFlag warns the user of region being deprecated in future versions. We should remove this flag
+// and only add region to commands which use it in the future.
+func DeprecateFlag(flags *pflag.FlagSet, reporter *reporter.Object) {
+	if flags.Changed(regionFlag) {
+		reporter.Warnf("Region flag will be deprecated in a future version for this command.")
```

Comment:
@hunterkepley We should ensure the two deprecation messages are the same. Shall we create a const and refer to that?

Reply:
Or perhaps:

```
flag := flags.Lookup("region")
if flag.Changed {
 reporter.Warnf(flag.Deprecated)
}
```